### PR TITLE
v2.0.0

### PR DESCRIPTION
**Pull request by jgravois**
checklist in #158 

- [x] ~~rest-js needs to `export { IEditFeatureResult }`~~
- [x] ~~`addResults?:` (et. al) need to _not_ be optional in responses.~~

AFFECTS PACKAGES:
@esri/hub-annotations
@esri/hub-auth
@esri/hub-common
@esri/hub-domains
@esri/hub-events
@esri/hub-initiatives
@esri/hub-sites
@esri/hub-solutions

### CODE REVIEW

**Comment by tomwayson** [comment]
Wow, this looks like it was a lot of work.

I'd like to merge and release #149 before merging this.

**Comment by jgravois** [comment]
> I'd like to merge and release #149 before merging this.

word.

**Comment by tomwayson** [comment]
As per our conversation this morning we should: 
- make the rest-js packages `peerDependencies` instead of `dependencies`
- decide on whether or not we are going to bump major w/ this, and if we do...
- ... what else would we change at that time?

**Comment by jgravois** [comment]
> make the rest-js packages `peerDependencies`

agreed.

> decide on whether or not we are going to bump major w/ this

it would be poor form not to.

> what else would we change at that time?

strangely enough, nothing comes to mind.

**Comment by tomwayson** [comment]
> strangely enough, nothing comes to mind.

I can think of a couple: #158

File: packages/annotations/src/add.ts [hunk 1]
```diff
@@ -3,11 +3,11 @@
...
   addFeatures,
-  IAddFeaturesRequestOptions,
-  IAddFeaturesResult
-} from "@esri/arcgis-rest-feature-service";
+  IAddFeaturesOptions
+} from "@esri/arcgis-rest-feature-layer";
 
-import { IFeature } from "@esri/arcgis-rest-common-types";
+import { IFeature } from "@esri/arcgis-rest-types";
+import { IEditFeatureResult } from "@esri/arcgis-rest-feature-layer/dist/esm/helpers";
```
Comment (tomwayson) [inline packages/annotations/src/add.ts]: is this not exported from `"@esri/arcgis-rest-feature-layer"`?

Comment (jgravois) [reply packages/annotations/src/add.ts]: it will be in rest-js `v2.0.1`

**Pull request merged**